pkg/sql/alter_column_type.go

```diff
@@ -111,10 +111,23 @@ func AlterColumnType(
 
 	var kind schemachange.ColumnConversionKind
 	if t.Using != nil {
-		// If an expression is provided, we always need to try a general conversion.
-		// We have to follow the process to create a new column and backfill it
-		// using the expression.
-		kind = schemachange.ColumnConversionGeneral
+		// If Using is provided and the experssion is identitcal to the new type cast,
+		// we use ClassifyConversion to determine the difficulty.
+		// Otherwise, we always need to try a general conversion and follow the process
+		// to create a new column and backfill it using the expression.
+		newColCast := tree.CastExpr{
+			Expr:       &tree.ColumnItem{ColumnName: tree.Name(col.GetName())},
+			Type:       typ,
+			SyntaxMode: tree.CastShort,
+		}
+		if tree.Serialize(&newColCast) == tree.Serialize(t.Using) {
```

Comment:
This should be improved to check the equality of the new cast and using expression.
For example, if the Using expression is not a CastShort, it will still have the same effect but skipped here.